scripts/run_portserver.py

```diff
@@ -410,7 +420,7 @@ class Server:
     max_backlog = 5
     message_size = 1024
 
-    def __init__(self, handler, socket_path):
+    def __init__(self, handler: Callable[..., Any], socket_path: str) -> None:
```

Comment:
....

Reply:
Something is indeed probably off here, could you please check with @U8NWXD (who wrote this code)? We shouldn't use Any.

Reply:
In practice, `handler` can accept any input that it can call `int()` on, which is why the tests can be so variable without failing. My guess is that the handler should accept bytes, but I'd recommend adding a print statement or a breakpoint to check what kind of data it receives in practice.

The handler definitely returns the allocated port number as a string though, not an int.
https://github.com/oppia/oppia/blob/fb0c69b8fcd531169b496e3341fb483cc2d9920a/scripts/run_portserver.py#L305

Reply:
> The handler definitely returns the allocated port number as a string though, not an int.

But In `handle_port_request`, we are returning Optional[bytes] not string. Because here we are returning bytes https://github.com/oppia/oppia/blob/fb0c69b8fcd531169b496e3341fb483cc2d9920a/scripts/run_portserver.py#L332
and here we are returning None https://github.com/oppia/oppia/blob/fb0c69b8fcd531169b496e3341fb483cc2d9920a/scripts/run_portserver.py#L326

Reply:
@U8NWXD @seanlip PTAL!

Reply:
Sorry -- this thread is still open from my perspective as a reviewer, but you'll need to get @U8NWXD's help on this (I'm not familiar with the code, I just think what is currently going on here "smells" a bit and should be tightened up).

Reply:
@sahiljoster32 Yes, I agree. Sorry my comment wasn't clear. I was trying to emphasize that the port number is not returned as an int, and I used "string" in the generic sense (which includes bit strings that Python calls `bytes`), not the Python `str` type. I think as written, the return type here should be `Union[bytes, str, None]` since we can also return an empty string. However, I agree with Sean, this code is suspect. I'd probably re-write it to return empty byte strings in the failure cases instead of a mix of `None` and `''`

Reply:
Hm, one question. It seems to me that the point of `handler` is to return an allocated port number (or possibly None, I'm not sure).

If this is the case, why can't we just force the handlers to return ints|None and do any necessary casting within the handler? It seems really odd to me to return an int as a string (especially if it can never be a string) and I think that we should just constrain the interface on our end. (I assume we have control over the handler code in all cases?)

Reply:
I think @U8NWXD we need your help again, PTAL!

Reply:
The port number has to be sent over the socket, so it'll have to be converted to bytes. See the socket documentation here: https://docs.python.org/3/library/socket.html#socket.socket.sendall

Reply:
Yup, but we can do that conversion at the interface, right? In our domain logic we should just represent it as what it is.